packages/docusaurus-theme-classic/src/theme/DocSidebar/index.js

```diff
@@ -48,7 +68,8 @@ function DocSidebarItem({item, onItemClick, collapsible, ...props}) {
             <a
               className={classnames('menu__link', {
                 'menu__link--sublist': collapsible,
-                'menu__link--active': collapsible && !item.collapsed,
+                'menu__link--active':
+                  collapsible && !item.collapsed && isCategoryOfActivePage(),
```

Comment:
Not sure if I love the function call here, might be better to create a new variable and assign it to a function call and then use it here. Open to other ideas too

Reply:
It'd be better to make the function a pure one and pass in the require parameters. That makes it easier to test and move around.

Reply:
Good call! I'll refactor that